pkg/scheduler/framework/plugins/interpodaffinity/scoring.go

```diff
@@ -209,29 +156,11 @@ func (pl *InterPodAffinity) PreScore(
 		}
 	}
 
```

Comment:
nit: L201 may be non-trivial, and hence that could be a waste when L205 is satisfied. How about switch to:

```go
	var allNodes []*framework.NodeInfo
	var err error
	// Unless the pod being scheduled has affinity terms, we only
	// need to process nodes hosting pods with affinity.
	if hasAffinityConstraints || hasAntiAffinityConstraints {
		allNodes, err = pl.sharedLister.NodeInfos().List()
		if err != nil {
			framework.NewStatus(framework.Error, fmt.Sprintf("get all nodes from shared lister error, err: %v", err))
		}
	} else {
		allNodes, err = pl.sharedLister.NodeInfos().HavePodsWithAffinityList()
		if err != nil {
			framework.NewStatus(framework.Error, fmt.Sprintf("get pods with affinity list error, err: %v", err))
		}
	}
```